work_queue/src/work_queue.c

```diff
@@ -6184,17 +6184,73 @@ struct work_queue_task *work_queue_wait_internal(struct work_queue *q, int timeo
 
 int work_queue_hungry(struct work_queue *q)
 {
-	int ready = task_state_count(q, NULL, WORK_QUEUE_TASK_READY);
-	if(q->stats->tasks_dispatched < 100)
-		return MAX(100 - ready, 0);
+	//check if queue is initialized
+	//return false if not
+	if (q == NULL){
+		return 0;
+	}
 
-	//BUG: fix this so that it actually looks at the number of cores available.
+	//check if queue's statistics are initialized
```

Comment:
Ah, I see the problem here.  `q->stats` is used internally to track some stats incrementally, but doesn't track those that are expensive to compute.  This is done in `work_queue_get_stats` which takes some things from `q->stats` and then adds the computed items.

So, what you want to do is this:
```
work_queue_stats s;
work_queue_get_stats(q,&s);
```

Then everything in `s` (like `tasks_waiting` will be up to date and can be used normally.